internal/extender/resource.go

```diff
@@ -106,15 +107,20 @@ func NewExtender(
 // Predicate is responsible for returning a filtered list of nodes that qualify to schedule the pod provided in the
 // ExtenderArgs
 func (s *SparkSchedulerExtender) Predicate(ctx context.Context, args schedulerapi.ExtenderArgs) *schedulerapi.ExtenderFilterResult {
-	params := internal.PodSafeParams(args.Pod)
+	params := internal.PodSafeParams(*args.Pod)
 	role := args.Pod.Labels[SparkRoleLabel]
-	instanceGroup := args.Pod.Spec.NodeSelector[s.instanceGroupLabel]
-	params["podSparkRole"] = role
-	params["instanceGroup"] = instanceGroup
 	ctx = svc1log.WithLoggerParams(ctx, svc1log.SafeParams(params))
 	logger := svc1log.FromContext(ctx)
+	instanceGroup, success := internal.FindInstanceGroupFromPodSpec(args.Pod.Spec, s.instanceGroupLabel)
+	if !success {
```

Comment:
I think we should preserve the previous behaviour that if the instance group node affinity doesn't exist, we treat it as `""` instead of failing

Reply:
done